File: Easy/Python/1572. Matrix Diagonal Sum.py
class Solution(object):
    def diagonalSum(self, mat):
        sum = 0
        top = 0
        bot = len(mat[0]) - 1

        # Subtract the value of any secondary diagonal
        if len(mat[0]) % 2 == 1:
            sum = sum - mat[bot / 2][bot / 2]
        
        # Add each value from the matrix diagonals
        for i in range(len(mat[0])):
            sum = sum + mat[i][top] + mat[i][bot]
            top += 1
            bot -= 1
        
        return sum

File: Easy/Java/1572. Matrix Diagonal Sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int sum = 0;
        int top = 0;
        int bot = mat[0].length - 1;

        // Subtract the value of any secondary diagonal
        if(mat[0].length % 2 == 1)
            sum -= mat[bot / 2][bot / 2];
        
        // Add each value from the matrix diagonals
        for(int i = 0; i < mat[0].length; i++){
            sum += mat[i][top] + mat[i][bot];
            top++;
            bot--;
        }

        return sum;
    }
}
